File: services/server/src/main/java/com/yaytsa/server/controller/TimeController.java
package com.yaytsa.server.controller;

import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.responses.ApiResponse;
import io.swagger.v3.oas.annotations.security.SecurityRequirements;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@Tag(name = "System", description = "System information and configuration")
public class TimeController {

  @Operation(
      summary = "Get server time",
      description = "Returns server timestamp in milliseconds for clock synchronization")
  @ApiResponse(responseCode = "200", description = "Server time in milliseconds as plain text")
  @SecurityRequirements
  @GetMapping(value = "/v1/time", produces = MediaType.TEXT_PLAIN_VALUE)
  public String getServerTime() {
    return Long.toString(System.currentTimeMillis());
  }
}

File: yay-tsa-v2/app/src/main/kotlin/dev/yaytsa/app/groups/TimeController.kt
package dev.yaytsa.app.groups

import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController
import java.time.Instant

@RestController
class TimeController {
    // ServerClock samples this to estimate clock offset/RTT; it reads the body as a bare number.
    @GetMapping("/v1/time", produces = [MediaType.TEXT_PLAIN_VALUE])
    fun serverTime(): String = Instant.now().toEpochMilli().toString()
}
